### DISCUSSION TITLE
NuxtJS sessions not persisting

### DISCUSSION THREAD

**Discussion by nezia1** [supabase/supabase#1737] [Questions]
Hello,
I've been trying to setup Supabase with NuxtJS but for some reason the session is not persisting between page reloads. I use the nuxt-supabase wrapper to use Supabase in my Nuxt context, but whenever I connect using supabase.auth.signIn, I have a Vuex action called checkAuth, which should return the current user session, but it always returns null. Any idea why is that? Is there any way to persist the Supabase authentication on Nuxt?

**Comment by zernonia** [comment]
Since NuxtJS utilize Vuex, I believe you can use [VueX Persisted State](https://www.npmjs.com/package/vuex-persistedstate) to allow VueX to persist between page reloads.

I hope this solve the problem. If not, I'm curious to see your `checkAuth` function
